source/vir/src/well_formed.rs

```diff
@@ -372,6 +373,17 @@ fn check_one_expr(
                 VisitorControlFlow::Stop(e) => Err(e),
             }?;
         }
+        ExprX::AssertBy { ensure, vars, .. } => match &ensure.x {
+            ExprX::Binary(crate::ast::BinaryOp::Implies, _, _) => {
+                if !vars.is_empty() {
+                    diags.push(VirErrAs::Warning(
+                        error(&expr.span, "using ==> in `assert forall` will not assume the antecedent in the body, and is deprecated, consier using `implies` instead of `==>`
+if you did not mean to assume the antecedent, please open an issue on the Verus issue tracker: `assert forall` will always assume the antecedent of the outermost implication in the future"),
```

Comment:
I found this somewhat confusing. How about this instead? "If you didn't mean to assume the antecedent, we're very curious to hear why! To tell us, please open an issue on the Verus issue tracker on github with the title `Don't always make assert forall assume the antecedent`." If no one opens such an issue, we'll soon change the behavior of Verus to always assume the antecedent of the outermost implication.

Reply:
By the way, I've started using it and it's working great. I found a case where I was using `assert forall |x| A ==> B ==> C`.

Reply:
I wrote an alternate version (https://github.com/verus-lang/verus/pull/1302#discussion_r1794440631) in the other comment thread before noticing @jaylorch's comment here.  I like the version with the "If no one opens ..." which makes the plan incredibly clear.